concourse/pipelines/guest-package-build.jsonnet

```diff
@@ -164,22 +257,79 @@ local promotepackagejob = {
   ] else [],
   on_success: {
     task: 'success',
-    file: 'guest-test-infra/concourse/tasks/publish-job-result.yaml',
-    vars: {
-      pipeline: 'guest-package-build',
-      job: tl.name,
-      result_state: 'success',
-      start_timestamp: '((.:start-timestamp-ms))',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'registry-image',
+        source: { repository: 'gcr.io/compute-image-tools/daisy' },
+      },
+      run: {
+        path: '/publish-job-result',
+        args: [
+          '--project-id=gcp-guest',
+          '--zone=us-west1-a',
+          '--pipeline=guest-package-build',
+          '--job=build-' + tl.package,
+          '--task=publish-job-result',
+          '--result-state=success',
+          '--start-timestamp=((.:start-timestamp-ms))',
+          '--metric-path=concourse/job/duration',
+        ],
+      },
     },
   },
   on_failure: {
     task: 'failure',
-    file: 'guest-test-infra/concourse/tasks/publish-job-result.yaml',
-    vars: {
-      pipeline: 'guest-package-build',
-      job: tl.name,
-      result_state: 'failure',
-      start_timestamp: '((.:start-timestamp-ms))',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'registry-image',
+        source: { repository: 'gcr.io/compute-image-tools/daisy' },
+      },
+      run: {
+        path: '/publish-job-result',
+        args: [
+          '--project-id=gcp-guest',
+          '--zone=us-west1-a',
+          '--pipeline=guest-package-build',
+          '--job=build-' + tl.package,
+          '--task=publish-job-result',
+          '--result-state=failure',
+          '--start-timestamp=((.:start-timestamp-ms))',
+          '--metric-path=concourse/job/duration',
+        ],
+      },
+    },
+  },
+};
+
+local uploadpackagetask = {
+  local tl = self,
+
+  package_paths:: error 'must set package_paths in uploadpackagetask',
+  repo:: error 'must set rapture_repo in uploadpackagetask',
+  topic:: 'projects/artifact-releaser-prod/topics/gcp-guest-package-upload-prod',
+  type:: 'uploadToStaging',
+  universe:: error 'must set universe in uploadpackagetask',
+
+  task: 'upload-' + tl.repo,
+  config: {
+    platform: 'linux',
+    image_resource: {
+      type: 'registry-image',
+      source: { repository: 'google/cloud-sdk', tag: 'alpine' },
+    },
+    run: {
+      path: 'gcloud',
```

Comment:
Is there a reason to why some cases you're using full path and others you aren't?

Reply:
yes just copying from our existing task files